core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -74,11 +74,20 @@ import scala.jdk.CollectionConverters._
 /*
  * Result metadata of a log append operation on the log
  */
-case class LogAppendResult(info: LogAppendInfo, exception: Option[Throwable] = None) {
+case class LogAppendResult(info: LogAppendInfo,
+                           exception: Option[Throwable],
+                           hasCustomErrorMessage: Boolean) {
```

Comment:
We could make the hasCustomErrorMessage param false by default, but there are pros and cons to it. 
I'm ok with keeping as is.

Reply:
honestly, I'm just not a fan of having default args. I think it makes things less explicit, hence the change.

Reply:
That's fair.